diff --git a/src/reuse/vcs.py b/src/reuse/vcs.py
index 8c1cb5c8edc3d94e..09fe5a70a653c69d 100644
--- a/src/reuse/vcs.py
+++ b/src/reuse/vcs.py
@@ -261,7 +261,7 @@ class VCSStrategyJujutsu(VCSStrategy):
         """
         Return a set of all files tracked in the current jj revision
         """
-        command = [str(self.EXE), "files"]
+        command = [str(self.EXE), "file", "list"]
         result = execute_command(command, _LOGGER, cwd=self.root)
         all_files = result.stdout.decode("utf-8").split("\n")
         return {Path(file_) for file_ in all_files if file_}
