diff --git a/src/rustdoc_gen.rs b/src/rustdoc_gen.rs
index eadb1f756ef951c8..d0156d9fa5379869 100644
--- a/src/rustdoc_gen.rs
+++ b/src/rustdoc_gen.rs
@@ -479,7 +479,9 @@ impl RustdocFromGitRevision {
         config: &mut GlobalConfig,
     ) -> anyhow::Result<Self> {
         config.shell_status("Cloning", rev)?;
-        let repo = gix::discover(source)?;
+        let repo: gix::Repository =
+            gix::ThreadSafeRepository::discover_with_environment_overrides(source)
+                .map(Into::into)?;

         let tree_id = repo.rev_parse_single(&*format!("{rev}^{{tree}}"))?;
         let tree_dir = target.join(tree_id.to_string());
